go/mysql/query.go

```diff
@@ -650,7 +650,11 @@ func (c *Conn) parseComStmtExecute(prepareData map[uint32]*PrepareData, data []b
 			}
 
 			// convert MySQL type to internal type.
-			valType, err := sqltypes.MySQLToType(int64(mysqlType), int64(flags))
+			// for com_stmt_execute, the flag will either be 0x00 or 0x80 indicating signed or unsigned
+			// as a result, we need to shift the flag to the right by 2 bits
+			// while this flag may conflict with the mysqlBinary constant, it doesn't matter;
+			// the format for BLOB and TEXT parameters is the same for both binary and non-binary
+			valType, err := sqltypes.MySQLToType(int64(mysqlType), int64(flags) >> 2)
```

Comment:
is `0x20 >> 2` the way we convert the flag const to `0x80`?

Reply:
The opposite, I'm converting `0x80` to `0x20`, so the `sqltypes.MySQLToType` logic can stay the same.

Reply:
oh i see, yeah this all makes sense to me now